geode-core/src/main/java/org/apache/geode/internal/net/NioSslEngine.java

```diff
@@ -373,6 +397,9 @@ public void doneReadingDirectAck(ByteBuffer unwrappedBuffer) {
 
   @Override
   public synchronized void close(SocketChannel socketChannel) {
+
+    assert socketChannel.isBlocking();
+
```

Comment:
Code inside this method (not modified by this PR):

```java
        while (myNetData.hasRemaining()) {
          socketChannel.write(myNetData);
        }
```

Writes in what would be a hot loop if the channel was in non-blocking mode. Rather than changing that loop, I added this assertion.